packages/loopring_v3/circuit/Gadgets/MathGadgets.h

```diff
@@ -539,12 +532,8 @@ class TernaryGadget : public GadgetT
         pb.val(selected) = (pb.val(b) == FieldT::one()) ? pb.val(x) : pb.val(y);
     }
 
-    void generate_r1cs_constraints(bool enforceBitness = true)
+    void generate_r1cs_constraints()
     {
-        if (enforceBitness)
-        {
-            libsnark::generate_boolean_r1cs_constraint<ethsnarks::FieldT>(pb, b, FMT(annotation_prefix, ".bitness"));
-        }
         pb.add_r1cs_constraint(ConstraintT(b, y - x, y - selected), FMT(annotation_prefix, ".b * (y - x) == (y - selected)"));
```

Comment:
This is quite dangerous. I'm pretty sure this check cannot be removed with how things are now, the circuits probably depend on this in some places.